File: Assets/ToLua/Lua/user/ui/shop/model/data/shop_diamond_data.lua
local t = {}
t.data = {}
local item = {}
item.__index = item
local game_res_data = require('ui/game/model/game_res_data')

function item.New(table)
	local o = {}
	setmetatable(o,item)
	o.id = tonumber(table.id)
	o.sheet_num = tonumber(table.sheet_num)
	o.buy_type = tonumber(table.buy_type)
	o.resource = table.resource
	o.name = table.name
	o.pic = table.pic
	o.des = table.des
	o.cost = table.cost
	o.first_award = game_res_data.NewByString(table.first_award)
  o.buy_award = game_res_data.NewByString(table.buy_award)
	o.double_base = tonumber(table.double_base)
  o.mounth_card = tonumber(table.mounth_card) > 0
  o.sort = tonumber(table.sort)
  o.ios_is_show = tonumber(table.ios_is_show) > 0
	o.name1 = table.name1
	o.des1 = table.des1
	return o
end
function t.GetDataById(id)
	return t.data[id]
end

local function Start()
	local origin = dofile('shop_diamond')
	origin.ForEach(function(id,table)
		local newItem = item.New(table)
		t.data[id] = newItem
	end)
end

Start()
return t

File: Assets/Code/Logic/Shop/Model/ShopDiamondData.cs
using System.Collections.Generic;
using Common.Util;
using Logic.Game.Model;

namespace Logic.Shop.Model
{
	public class ShopDiamondData
	{
		private static Dictionary<int, ShopDiamondData> _shopDiamondDataDictionary;
		public static Dictionary<int, ShopDiamondData> GetShopDiamondData ()
		{
			if (_shopDiamondDataDictionary == null)
			{
				_shopDiamondDataDictionary = CSVUtil.Parse<int, ShopDiamondData>("config/csv/shop_diamond", "id");
			}
			return _shopDiamondDataDictionary;
		}

        public static ShopDiamondData GetShopDiamondDataById(int id) 
        {
            if (_shopDiamondDataDictionary == null)
                GetShopDiamondData();
            return _shopDiamondDataDictionary[id];
        }

		[CSVElement("id")]
		public int id;

		[CSVElement("sheet_num")]
		public int shopID;

		[CSVElement("buy_type")]
		public int buyType;

		[CSVElement("resource")]
		public string resource;

		[CSVElement("name")]
		public string name;

        [CSVElement("name1")]
        public string name1;

		[CSVElement("pic")]
		public string pic;

		[CSVElement("des")]
		public string description;

		public GameResData costGameResData;
		[CSVElement("cost")]
		public string cost
		{
			set
			{
				costGameResData = new GameResData(value);
			}
		}

		public GameResData firstAwardGameResData;
		[CSVElement("first_award")]
		public string first_award
		{
			set
			{
//				firstAwardGameResData = new GameResData(value);
			}
		}

		public GameResData buyAwardGameResData;
		[CSVElement("buy_award")]
		public string buy_award
		{
			set
			{
//				buyAwardGameResData = new GameResData(value);
			}
		}

		[CSVElement("double_base")]
		public int double_base;
	}
}
